gen_future_weather_file.py

```diff
@@ -0,0 +1,425 @@
+
+        Gets a list of data for each day of each year within time_range from
+        the climate file where the location is closest to lat, long.
+
+        Args:
+            climate_file(str): The path to the climate file to read from.
+            lat(float): The latitude to read data from.
+            long(float): The longitude to read data from.
+            cdfvariable(str): The variable to read from the netcdf file.
+            time_range(range): The range of years to read data from, to.
+
+        Returns:
+            a list with 366 lists. One for each day of the year in a leap year.
+            Each list contains all the data found in the file for that
+            specific day, with each entry in the inner list is 1 year's entry
+            for that day.
+    """
+
+    # Read the climate data out of the climate file.
+    raw_data = Dataset(climate_file)
+
+    # Get a list of the dates in the climate file.
+    data_dates = cdf.num2date(raw_data["time"][:], raw_data["time"].units)
+
+    # Get the latitude of each location in the file.
+    lat_data = raw_data.variables["lat"][:]
+
+    # Get the logitude of each location in the file.
+    long_data = raw_data.variables["lon"][:]
+
+    # Find the incides of the data with the closest lat and long to
+    # those passed.
+    lat_index = lat_data.tolist().index(
+                                  lat_data[np.abs(lat_data - lat).argmin()]
+                                  )
+
+    long_index = long_data.tolist().index(
+                                  long_data[np.abs(long_data - long).argmin()]
+                                  )
+
+    # Grab the actual relevant data from the file.
+    data = [[] for _ in range(366)]
+    for index, datapoint in enumerate(raw_data.variables[cdfvariable]):
+        time = data_dates[index]
+
+        if time.year in time_range:
+
+            # Appends the datapoint for the lat and long to its day's index in
+            # the list.
+            # FIXME: leap year issue.
+            #
+            #    It is probable that time.timetuple().tm_yday wont return 366
+            #    for Feb 29th. If this is the case then a new function needs to
+            #    be written to replace time.timetuple().tm_yday that returns
+            #    366 for Feb 29th, and all occurrences of
+            #    time.timetuple().tm_yday need to be replaced with that
+            #    function. This is an issue because if feb 29th is not day 366,
+            #    then all days after feb 29th in a leap year will be shifted by
+            #    one, causing issues with averaged etc.
+            data[time.timetuple().tm_yday - 1].append(
+                                              datapoint[lat_index][long_index]
+                                               )
+
+    return data
+
+
+def avg(lst: list):
+    """ This method returns the average value of a passed list."""
+
+    return sum(lst) / len(lst) if len(lst) != 0 else 0
```

Comment:
This isn't a good idea. The average of a zero-length list is undefined (NaN), not zero. This has the bad potential to hide errors in the data and propagate badness.